server/datastore/mysql/android_test.go

```diff
@@ -357,3 +359,99 @@ func testAndroidHostStorageData(t *testing.T, ds *Datastore) {
 	assert.Equal(t, 64.0, updatedFullHost.GigsDiskSpaceAvailable, "Updated available disk space should be saved in host_disks")
 	assert.Equal(t, 25.0, updatedFullHost.PercentDiskSpaceAvailable, "Updated disk space percentage should be saved in host_disks")
 }
+
+func testNewAndroidHostWithIdP(t *testing.T, ds *Datastore) {
+	test.AddBuiltinLabels(t, ds)
+
+	// create IdP account... InsertMDMIdPAccount generates its own UUID
+	idpAccount := &fleet.MDMIdPAccount{
+		Username: "john.doe",
+		Fullname: "John Doe",
+		Email:    "john.doe@example.com",
+	}
+	err := ds.InsertMDMIdPAccount(context.Background(), idpAccount)
```

Comment:
This is fine as-is so feel free to ignore, but one pattern we've used a lot is to define a `ctx := t.Context()` at the beginning of the test function, when we need to pass a context in many places.